src/pages/lesson-details/LessonDetails.tsx

```diff
@@ -78,6 +99,65 @@ const LessonDetails = () => {
     })
   }
 
+  const getCooperation = useCallback(() => {
+    return cooperationService.getCooperationById(id)
+  }, [id])
+
+  const { data: cooperation } = useQuery({
+    queryFn: getCooperation,
+    queryKey: ['cooperation', id]
+  })
+
+  useEffect(() => {
+    if (!cooperation?.sections) return
+
+    cooperation.sections?.forEach((section) => {
+      const resource = section.resources?.find(
+        (resource) => resource.resource._id === lessonId
+      )
+
+      if (resource) {
+        setCompletionStatus(
+          resource.completionStatus ?? CompletionStatusEnum.Completed
+        )
+      }
+    })
+  }, [cooperation, lessonId, setCompletionStatus])
+
+  const handleResponseError = useCallback(
+    (error: ResponseError) => {
+      dispatch(
+        openAlert({
+          severity: snackbarVariants.error,
+          message: getErrorKey(error)
+        })
+      )
+    },
+    [dispatch]
+  )
+
+  const handleResponse = () => {
+    void queryClient.invalidateQueries({ queryKey: ['lesson', lessonId] })
+    void queryClient.invalidateQueries({ queryKey: ['cooperation', id] })
+  }
+
+  const { mutate: updateLessonStatus } = useMutation({
+    mutationFn: cooperationService.updateResourceCompletionStatus,
+    onSuccess: handleResponse,
+    onError: handleResponseError
+  })
```

Comment:
There is a `queryKeys` property, use it please

```suggestion

  const { mutate: updateLessonStatus } = useMutation({
    mutationFn: cooperationService.updateResourceCompletionStatus,
    queryKeys: [['lesson', lessonId], ['cooperation', id]],
    onError: handleResponseError
  })
```

Reply:
fixed